pkgdb/src/buildenv/realise.cc

```diff
@@ -60,59 +56,397 @@ namespace flox::buildenv {
+    # as configuration statements by the "in-place" activation
+    # mode. So, we'll redirect stdout to stderr.
+    source "$FLOX_ENV/activate.d/hook-on-activate" 1>&2
+  fi
+
+  # Capture ending environment.
+  _end_env="$($_coreutils/bin/mktemp --suffix=.$FLOX_ENV_PID.end-env)"
+  export | $_coreutils/bin/sort > "$_end_env"
+
+  # The userShell initialization scripts that follow have the potential to undo
+  # the environment modifications performed above, so we must first calculate
+  # all changes made to the environment so far so that we can restore them after
+  # the userShell initialization scripts have run. We use the `comm(1)` command
+  # to compare the starting and ending environment captures (think of it as a
+  # better diff for comparing sorted files), and `sed(1)` to format the output
+  # in the best format for use in each language-specific activation script.
+  _add_env="$($_coreutils/bin/mktemp --suffix=.$FLOX_ENV_PID.add-env)"
+  _del_env="$($_coreutils/bin/mktemp --suffix=.$FLOX_ENV_PID.del-env)"
+
+  # Export tempfile paths for use within shell-specific activation scripts.
+  export _add_env _del_env
+
+  # Capture environment variables to _set_ as "key=value" pairs.
+  $_coreutils/bin/comm -13 "$_start_env" "$_end_env" | \
+    $_gnused/bin/sed -e 's/^declare -x //' > $_add_env
+
+  # Capture environment variables to _unset_ as a list of keys.
+  # TODO: remove from $_del_env keys set in $_add_env
+  $_coreutils/bin/comm -23 "$_start_env" "$_end_env" | \
+    $_gnused/bin/sed -e 's/^declare -x //' -e 's/=.*//' > $_del_env
+
+  # Don't need these anymore.
+  $_coreutils/bin/rm -f "$_start_env" "$_end_env"
+
+else
+
+  # "Reactivation" of this environment.
+
+  # If we're attempting to launch an interactive shell then just print a
+  # message to say that the environment has already been activated.
+  if [ -t 1 ] && [ $# -eq 0 ]; then
+    echo "ERROR: Flox environment already activated: $FLOX_ENV" >&2
+    exit 1
+  fi
+
+  # Assert that the expected _{add,del}_env variables are present.
+  [ -n "$_add_env" -a -n "$_del_env" ] || {
+    echo 'ERROR (activate): $_add_env and $_del_env not found in environment' >&2;
+    if [ -h "$FLOX_ENV" ]; then
+      echo "moving $FLOX_ENV link to $FLOX_ENV.$$ - please try again" >&2;
+      $_coreutils/bin/mv $FLOX_ENV $FLOX_ENV.$$
+    fi
+    exit 1;
+  }
+
+  # Replay the environment for the benefit of this shell.
+  eval "$($_gnused/bin/sed -e 's/^/unset /' $_del_env)"
+  eval "$($_gnused/bin/sed -e 's/^/export /' $_add_env)"
+
+fi
+
+# From this point on the activation process depends on the mode:
+
+# 1. "command" mode(s): invoke the user's shell with args that:
+#   a. defeat the shell's normal startup scripts
+#   b. source the relevant activation script
+#   c. invoke the command in one of "stdin" or "-c" modes
+if [ $# -gt 0 ]; then
+  if [ $# -ne 2 -o "$1" != "-c" ]; then
+    # Marshal the provided args into a single safely-quoted string.
```

Comment:
```suggestion
    # Marshall the provided args into a single safely-quoted string.
```

Reply:
From https://www.merriam-webster.com/grammar/laying-down-the-law-on-martial-and-marshal-and-marshall 😄 

### Marshal as a Verb

Marshals don't always marshal, but when you want a verb, it's marshal and not martial you're after. As a verb, [marshal](https://www.merriam-webster.com/dictionary/marshal#h2) usually means:
...
> to arrange or prepare (something, such as your thoughts or ideas) in a clear, effective, or organized way, e.g., marshaling an argument